### PR TITLE
Fixes: #1673 BUG Tooltip not visible when more than one row ellipsis is applied to Text component 

### PR DESCRIPTION
**Pull request by viditagrawal56**
Fixes : #1673 

#### Basic

- [ ] Used plop (`npm run plop`) to create a new component.
- [X] PR has description.
- [X] New component is functional and uses Hooks.
- [X] Component is written in TypeScript.

#### Style

- [ ] Styles are added to `NewComponent.modules.scss` file inside the `NewComponent` folder.
- [ ] Component uses CSS Modules.
- [ ] Design is compatible with [Monday Design System](https://design.monday.com/).
- [ ] Styles are defined using [monday-ui-style](https://github.com/mondaycom/monday-ui-style) tokens
- [ ] Displays correctly in all the themes

#### Storybook

- [ ] Stories were added to `/src/components/NewComponent/__stories__/NewComponent.stories.mdx` file.
- [ ] Stories include all flows of using the component.
- [ ] Stories implemented using [vibe-storybook-components](https://github.com/mondaycom/vibe-storybook-components) components and tokens.

#### Tests

- [ ] Tests are compliant with [TESTING_README.md](TESTING_README.md) instructions.

#### Accessibility

- [ ] Component is accessible.
- [ ] Component is compatible with [Vibe Accessibility Guidelines](https://style.monday.com/?path=/docs/foundations-accessibility--page)

### CODE REVIEW

**Comment by viditagrawal56** [comment]
Hey @SergeyRoyt, I understand that you must be busy,
I made this draft PR
Please let me know if any changes are required! :)

**Comment by viditagrawal56** [comment]
> Hey @SergeyRoyt, I understand that you must be busy that's why you were not replying to my messages 😅. I made this draft PR, Please let me know if any changes are required whenever you are free! :)

@SeregyRoyt

File: src/components/Text/utils/TextContext.ts
Comment (SergeyRoyt) [inline src/components/Text/utils/TextContext.ts:1]: Let's call this one `TypographyContext` and move it `Typography` folder, cause it's related to Typography, not just to Text

Comment (SergeyRoyt) [reply src/components/Text/utils/TextContext.ts:1]: Let's fix lint issues here

File: src/components/Text/Text.tsx [hunk 1]
```diff
@@ -32,9 +33,11 @@ const Text: VibeComponent<TextProps, HTMLElement> & {
       ...typographyProps
     },
     ref
-  ) => {
+    ) => {
+    const { ignoreHeightOverflow } = useContext(TextContext);
```
Comment (SergeyRoyt) [inline src/components/Text/Text.tsx:37]: That's not right. You should call `useContext` in the `Typography` component, then pass this value into `useTooltipProps` which should pass it further to `useIsOverflowing` hook

File: src/components/Text/Text.tsx [hunk 2]
```diff
@@ -32,9 +33,11 @@ const Text: VibeComponent<TextProps, HTMLElement> & {
...
     const overrideEllipsis = ellipsis ?? element !== "p";
     return (
+      <TextContext.Provider value={{ignoreHeightOverflow}}>
```
Comment (SergeyRoyt) [inline src/components/Text/Text.tsx:40]:
```suggestion
      <TypographyContext.Provider value={{ignoreHeightOverflow: false}}>
```
And to make it explicit let's also provide TypographyContext value in `Heading` component, but there it should be 
`{ignoreHeightOverflow: true}` for now

**Review by SergeyRoyt — CHANGES_REQUESTED** [review]
Hey, @viditagrawal56 !
Sorry for not a very quick respond!
Wrote you a few notes, feel free to ask if you need help with anything

Basically what I want us to achieve here is passing different kind of context from `Text` (`{ignoreHeightOverflow: false}`) and `Heading` (`{ignoreHeightOverflow: true}`). Then consuming this context in `Typography` component and make difference in overflowing logic due to it. 

Basically you could have achieved the same just by using props, but what we do here is most likely a temp fix (some work will still need to be done in `Heading` component to fix overflow for the case of multiple lines), so we don't want to spoil our components API by adding new props - so we just pass them kinda "behind the scenes"

**Comment by viditagrawal56** [comment]
@SergeyRoyt Sorry for the mistakes 🙏 
I have made all the changes as per your notes
Let me know if more changes are needed!
Forgot to turn on prettier last time 😅, but all the lint issues are fixed now.

I think I also need to update the snapshots by running `npm run test:update` right?

File: src/components/Text/Text.tsx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import React, { forwardRef, ReactNode } from "react";
+import React, { forwardRef, ReactNode, useContext } from "react";
```
Comment (SergeyRoyt) [inline src/components/Text/Text.tsx:1]: Only one lint issue left to fix

**Comment by SergeyRoyt** [comment]
Looks great, exactly what's needed to be done! 💪
Let me just test it on our production environment to make sure we are not breaking anything 🙏

**Review by SergeyRoyt — APPROVED** [review]
Approved! Thanks a lot for the great job, @viditagrawal56!! 👑👑
